spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraSQLContext.scala

```diff
@@ -37,6 +41,152 @@ class CassandraSQLContext(sc: SparkContext) extends SQLContext(sc) {
   @transient
   val conf = sc.getConf
 
+  @transient
+  private val clusterReadConf: Map[String, SparkConf] = Map()
+  @transient
+  private val keyspaceReadConf: Map[Seq[String], SparkConf] = Map()
+  @transient
+  private val tableReadConf: Map[Seq[String], SparkConf] = Map()
+
+  /** Add table level read configuration settings */
+  def addTableReadConf(keyspace: String, table: String, conf: SparkConf, cluster: Option[String]) = {
+    cluster match {
+      case Some(c) =>
+        if (StringUtils.isEmpty(c)) {
+          throw new IllegalArgumentException("cluster name can't be null or empty")
```

Comment:
Can you somehow abstract this validation? This fragment is repeated multiple times - why don't you just create a method to validate and maybe throw an exception?